# Домашнє завдання до теми "Python Basics"
В цьому ДЗ важливою умовою виконання завдань є те, що ми не імпортуємо жодних бібліотек, а робимо все на чистому Python, використовуючи лише базовий функціонал. Це необхідно аби відпрацювати роботу з базовими типами даних. А використовувати бібліотеки ми ще навчимось далі :)

## Робота з числами та арифметичними операторами

1.Створіть змінну, яка буде зберігати кількість яблук. І надайте їй значення 5. Виведіть значення змінної на екран.

In [10]:
apple = 5

2.Виведіть тип змінної, створеної в завданні (1).

In [11]:
display(apple)

5

3.Додайте до значення змінної, створеної в завданні (1) число `12` і виведіть результат на екран з допомогою команди `print()`.

In [12]:
apple += 12
print(apple)

17


4.Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [14]:
a = 101
b = 210
c = 111
d = 402
print((a + b) * c/d)

85.8731343283582


Очікувана відповідь: 85.8731343283582

5.Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [15]:
11283%89

69

6.Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [18]:
11283 // 89

126

# Робота з рядками

7.Запишіть у змінну `name` ваше імʼя.

In [20]:
name = "Olga"

8.Запишіть у змінну `surname` ваше прізвище.

In [21]:
surname = "Tereshchenko"

9.Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [34]:
print(name, surname)

name_surname = name + " " + surname
print(name_surname)

Olga Tereshchenko
Olga Tereshchenko


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva' 

10.Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [35]:
name_surname.upper()

'OLGA TERESHCHENKO'

Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

11.Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [36]:
name_surname[2]

'g'

Очікувана відповідь (тільки з вашими даними): 'n'

12.Виведіть першу та останню літеру вашого прізвища.

In [45]:
name_surname[0]

'O'

In [48]:
name_surname[-1]

'o'

Очікувана відповідь (тільки з вашими даними): ('P', 'a')

13.Виведіть довжину вашого прізвища.

In [47]:
len(surname)

12

Очікувана відповідь (тільки з вашими даними): 7

14.Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [15]:
city = 'Antananarivo'
quantity_a = city.count('a')
quantity_A = city.count('A')
quantity = quantity_a + quantity_A
quantity

4

Очікувана відповідь: 4

15.Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [1]:
my_string = "The quick brown fox jumps over the lazy dog!!!"

16.Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [4]:
my_string.split()

['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!']

Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`

In [16]:
"_".join(['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!'])

'The_quick_brown_fox_jumps_over_the_lazy_dog!!!'

# Робота зі колекціями

У нас є наступний список:

In [2]:
my_list = [
    78, 12, 14, 27, 47, 28, 51, 46, 94, 37, 42, 20, 52, 33, 11, 11, 51,
    20, 81, 54, 86, 51, 38, 42, 49, 99, 76, 17,  9, 23, 97, 29, 29, 63,
    40, 11, 31, 76, 15, 21, 29, 75, 35,  2, 10, 71, 17, 10, 13,  7, 44,
    12, 19, 61, 13, 25, 62, 46, 10, 44, 75, 57, 20, 57,  3, 44, 16, 28,
    79, 57, 62, 70, 95,  8, 47, 82, 80, 45, 36, 44, 79, 95, 73, 42, 29,
    83, 80, 66, 56,  8, 73,  5, 99, 65, 65, 98, 17, 36, 16, 60
]

Будемо з ним працювати в наступних завданнях.

17.Виведіть 5 перших елементів списку `my_list` з допомогою слайсингу.

In [3]:
my_list[0:5]

[78, 12, 14, 27, 47]

Очікувана відповідь: [78, 12, 14, 27, 47]

18.Виведіть 3 останніх елементи списку `my_list` з допомогою слайсингу.

In [9]:
my_list[-3:]

[36, 16, 60]

19.Відсортуйте за зростанням список `my_list`, запишіть результат у змінну `my_list_sorted` та виведіть значення нової змінної на екран.

In [12]:
my_list_sorted = my_list
my_list_sorted.sort()
my_list_sorted

[2,
 3,
 5,
 7,
 8,
 8,
 9,
 10,
 10,
 10,
 11,
 11,
 11,
 12,
 12,
 13,
 13,
 14,
 15,
 16,
 16,
 17,
 17,
 17,
 19,
 20,
 20,
 20,
 21,
 23,
 25,
 27,
 28,
 28,
 29,
 29,
 29,
 29,
 31,
 33,
 35,
 36,
 36,
 37,
 38,
 40,
 42,
 42,
 42,
 44,
 44,
 44,
 44,
 45,
 46,
 46,
 47,
 47,
 49,
 51,
 51,
 51,
 52,
 54,
 56,
 57,
 57,
 57,
 60,
 61,
 62,
 62,
 63,
 65,
 65,
 66,
 70,
 71,
 73,
 73,
 75,
 75,
 76,
 76,
 78,
 79,
 79,
 80,
 80,
 81,
 82,
 83,
 86,
 94,
 95,
 95,
 97,
 98,
 99,
 99]

20.Виведіть максимальний і мінімальний елемент у списку `my_list`. Сформуйте відповідь у наступному форматі
```
Max element  <число>
Min element  <число>
```
Можна скористатись результатом з попереднього завдання, або спеціальними фукнціями.

In [13]:
max(my_list), min(my_list)

(99, 2)

Очікувана відповідь: 

```
Max element  99
Min element  2
```

21.Створіть новий список `my_list_2`, який містить кожен 11-й елемент списку `my_list`. Виведіть `my_list_2` на екран.  
Памʼятаємо, що перший з "кожного одинадцятого елементу" списку має індекс 10.
Для виконання завдання використайте слайсинг.

In [ ]:
---

Очікуваний результат: `[42, 51, 29, 2, 13, 44, 80, 66, 16]`

22.Заданий список `new_list`. Перетворіть його на набір (set). Які зміни у наборі елементів Ви помітили? Відповідь напишіть у коментарі до коду (саме в клітинці з кодом). 

In [21]:
new_list = [22, 11, 67, 11, 88, 3213, 12, 78, 67, 11, 16, 15, 88]

new_set = set(new_list)
new_set #має лише унікальні елементи, не повторювальні

{11, 12, 15, 16, 22, 67, 78, 88, 3213}

23.Аби отримати скільки разів зустрічається деякий елемент в списку, зручно використовувати метод count(). Працює це наступним чином:
```
some_list = [1,1,1,2,2]
some_list.count(1)
# 3
```

Виведіть, скільки разів зустрічається число 11 у списку `new_list`.

In [23]:
new_list.count(11)

3

24.Наостанок, попрактикуємось працювати зі словниками. Це завдання включає 4 невеличких пункти:

1. Створіть словник, що містить 5 пар ключ-значення, де ключі - це імена студентів (типу рядок), а значення - їхні оцінки з програмування від 1 до 12. Запишіть словник у змінну.
2. Виведіть весь словник на екран. Важливо, аби словник вивівся на екран в jupyter notebook, будь-який спосіб, який це робить, нам підходить.
3. Додайте до словника нового студента та його оцінку.
4. Виведіть оцінку одного конкретного студента, звертаючись до неї за його ім'ям.

In [28]:
students = {
    "Victor": 4,
    "Emilie": 5,
    "Entony": 3,
    "Kate": 2
}
students['Tiffany'] = 12
students


{'Victor': 4, 'Emilie': 5, 'Entony': 3, 'Kate': 2, 'Tiffany': 12}

In [29]:
students['Tiffany']

12

# Ураа!! Ви завершили перше домашнє завдання!!!  
Аби здати ДЗ - залийте ноутбук у гітхаб репозиторій, який ви створили у перші дні навчання на курсі, та додайте посилання на файл у відповідному завданні на навчальній платформі.
Так тримати! 🚀🚀🚀